packages/go_router/lib/src/go_router_delegate.dart

```diff
@@ -441,30 +425,31 @@ class GoRouterDelegate extends RouterDelegate<Uri>
     ).toList();
 
     assert(matchStacks.isNotEmpty, 'no routes for location: $location');
-    assert(() {
-      if (matchStacks.length > 1) {
-        final StringBuffer sb = StringBuffer()
-          ..writeln('too many routes for location: $location');
-
-        for (final List<GoRouteMatch> stack in matchStacks) {
-          sb.writeln(
-              '\t${stack.map((GoRouteMatch m) => m.route.path).join(' => ')}');
-        }
-
-        assert(false, sb.toString());
-      }
-
-      assert(matchStacks.length == 1);
-      final GoRouteMatch match = matchStacks.first.last;
-      final String loc1 = _addQueryParams(match.subloc, match.queryParams);
-      final Uri uri2 = Uri.parse(location);
-      final String loc2 = _addQueryParams(uri2.path, uri2.queryParameters);
-
-      // NOTE: match the lower case, since subloc is canonicalized to match the
-      // path case whereas the location can be any case
-      assert(loc1.toLowerCase() == loc2.toLowerCase(), '$loc1 != $loc2');
-      return true;
-    }());
+    // Fixes https://github.com/flutter/flutter/issues/99833
```

Comment:
Instead of commenting out the code, can you add some explanation about whe it return first and remove the assert entirely

Reply:
@chunhtai  Added the explanation and removed assert

Reply:
Also added test to verify the it does not throw if there are multiple matches, and fixed code format